substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfInfoSectionImpl.java

```diff
@@ -724,18 +724,13 @@ private int writeMethodParameterDeclarations(DebugContext context, ClassEntry cl
         if (!Modifier.isStatic(range.getModifiers())) {
             pos = writeMethodParameterDeclaration(context, "this", classEntry.getTypeName(), true, isSpecification, buffer, pos);
         }
-        String paramsString = range.getParamSignature();
-        if (!paramsString.isEmpty()) {
-            String[] paramTypes = paramsString.split(",");
-            for (int i = 0; i < paramTypes.length; i++) {
-                String paramName = uniqueDebugString("");
-                String paramTypeName = paramTypes[i].trim();
-                FileEntry fileEntry = range.getFileEntry();
-                if (fileEntry != null) {
-                    pos = writeMethodParameterDeclaration(context, paramName, paramTypeName, false, isSpecification, buffer, pos);
-                } else {
-                    pos = writeMethodParameterDeclaration(context, paramTypeName, paramTypeName, false, isSpecification, buffer, pos);
-                }
+        for (String paramTypeName : range.getParamTypeNames()) {
+            String paramName = uniqueDebugString("");
+            FileEntry fileEntry = range.getFileEntry();
+            if (fileEntry != null) {
+                pos = writeMethodParameterDeclaration(context, paramName, paramTypeName, false, isSpecification, buffer, pos);
```

Comment:
Hmm, this if/else sequence is wrong -- but not because of anything you have done. Passing paramTypeName in place of paramName in the else branch makes no sense at all (paramName is ignored for now). The method call here should include a filename argument or null. This argument should drive the choice of abbrev code as either DW_ABBREV_CODE_method_parameter_declaration2 or DW_ABBREV_CODE_method_parameter_declaration3. It does not matter too much as we don't have a line or column number so a file name for the parameter declaration is not a lot of use. Still it needs fixing, especially as the current code makes no sense at all. I'll raise a separate issue for this.